### PR TITLE
GitHub Actions to build installation packages 

### PR DESCRIPTION
**Pull request by markrtuttle**
This pull request defines GitHub Actions to build installation packages for MacOS, Windows, and Ubuntu from the tip of the develop branch.  A new set of packages is generated each time a new commit is pushed to the develop branch.  The packages reside on GitHub as GitHub artifacts.  A web page like https://markrtuttle.github.io/cbmc-packages.html can be used to access and download the packages.

The purpose of this package is to complement the installation packages now available for MacOS and Ubuntu (eg, brew install cbmc), and also to generate packages to upload to a PPA when a new version is released.

See the README.md for more information.

The original Debian packaging was done by Michael Tautschnig.

### CODE REVIEW

**Review by NlightNFotis — COMMENTED** [review]
On a first pass over this, this looks good to me. 

I will do a second pass as I become more familiar with github actions, and approve once I feel confident this works as I fully expect it to.

**Review by NlightNFotis — APPROVED** [review]
I've done a second pass over it, and it looks solid.

File: .github/workflows/debian/surefire [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Origin: <vendor|upstream|other>, <url of original patch>
+Bug: <url in upstream bugtracker>
+Bug-Debian: https://bugs.debian.org/<bugnumber>
+Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
+Forwarded: <no|not-needed|url proving that it has been forwarded>
+Reviewed-By: <name and email of someone who approved the patch>
+Last-Update: 2018-09-27
+
+--- cbmc-5.10.orig/java-models-library/pom.xml
++++ cbmc-5.10/java-models-library/pom.xml
```
Comment (peterschrammel) [inline .github/workflows/debian/surefire:23]: Why can't apply this patch to the code base right away?

**Review by peterschrammel — APPROVED** [review]

**Comment by hannes-steffenhagen-diffblue** [comment]
@markrtuttle Can you check if the patch @peterschrammel commented on is still necessary, and if so can we work it into cbmc straight up instead? Otherwise I think this is good to go.

**Comment by markrtuttle** [comment]
@hannes-steffenhagen-diffblue, @peterschrammel, @tautschnig

I think this is ready to go.  I've added the fix to the debian builds that @tautschnig and I worked out.  Concerning the surefire patch that @peterschrammel and @hannes-steffenhagen-diffblue asked about, the debian build without this patch fails on Ubuntu16 and succeeds on Ubuntu18, so I'm leaving it in place because the Makefile for the debian builds is already complicated enough.

May I have a contact on the DiffBlue team to work out remaining issues? I have plans for how to store the packages built by GitHub Actions on S3, and how to advertise them.  I'd like to work out that plan before implementing.

**Comment by markrtuttle** [comment]
Recording the Ubuntu16 failure when the surefire patch is removed.

```
[WARNING] The POM for org.apache.maven.plugins:maven-surefire-plugin:jar:2.12.4 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.349 s
[INFO] Finished at: 2020-06-30T13:31:13Z
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.apache.maven.plugins:maven-surefire-plugin:2.12.4 or one of its dependencies could not be resolved: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact org.apache.maven.plugins:maven-surefire-plugin:jar:2.12.4 has not been downloaded from it before. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
cp: cannot stat '../../../lib/java-models-library/target/core-models.jar': No such file or directory
make[5]: *** [library] Error 1
Makefile:29: recipe for target 'library' failed
make[4]: *** [library] Error 2
make[5]: Leaving directory '/tmp/cbmc-5.12/jbmc/src/java_bytecode/library'
make[4]: *** Waiting for unfinished jobs....
Makefile:77: recipe for target 'library' failed
make[4]: Leaving directory '/tmp/cbmc-5.12/jbmc/src/java_bytecode'
Makefile:30: recipe for target 'java_bytecode.dir' failed
make[3]: Leaving directory '/tmp/cbmc-5.12/jbmc/src'
make[3]: *** [java_bytecode.dir] Error 2
debian/rules:44: recipe for target 'override_dh_auto_build' failed
make[2]: Leaving directory '/tmp/cbmc-5.12'
make[2]: *** [override_dh_auto_build] Error 2
make[1]: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debian/rules:22: recipe for target 'build' failed
make[1]: Leaving directory '/tmp/cbmc-5.12'
Makefile:255: recipe for target 'build' failed
make: *** [build] Error 2
make: Leaving directory '/tmp'
##[error]Process completed with exit code 2.
```

**Pull request merged**